Source/Apple/arcana/tracing/trace_region.h

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//
+
+#pragma once
+
+#include <os/signpost.h>
+
+namespace arcana
+{
+    class trace_region final
+    {
+    public:
+        trace_region() = delete;
+        trace_region& operator=(const trace_region&) = delete;
+        trace_region(const trace_region&) = delete;
+        trace_region& operator=(trace_region&&) = delete;
```

Comment:
Why have a move constructor but not move assignment? Seems like we should either implement move semantics completely or not.

Reply:
I can't imagine a scenario where you would use the move assignment operator without it being a programming error. Do you have something in mind?

Reply:
I don't know if I have a good one, but maybe if someone wanted to measure the lifetime of a movable RAII object, it would be stored in a member.